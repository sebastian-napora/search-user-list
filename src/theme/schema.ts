import { MEDIA_QURIES } from './mediaQueries';

import { TThemes } from './types';

const Theme = {
  data: 'main',
  id: 'T_001',
  name: 'Main',
  colors: {
    fonts: {
      primary: '#f1f1f1',
      white: '#ffffff' as const,
      grey: '#838383',
      greySecondary: '#a6a6a6'
    },
    white: '#ffffff' as const,
    body: '#FFFFFF',
    primary: '#490e4b',
    secondary: '#3b0044',
    greyPrimary: '#cfcfcf',
    greySecondary: '#e1dcdb',
    hoverColor: '#f5f5f5' as const,
    black: '#020000',
    text: '#000000',
    label: '#838383',
    button: {
      text: '#FFFFFF',
      background: '#000000'
    },
    link: {
      text: '#3b0044',
      opacity: 1
    }
  },
  mediaQueries: MEDIA_QURIES,
  font: {
    text: 'AauxPro OT'
  }
};

//changed in the future this is only example
export const THEMES: TThemes = {
  data: {
    main: Theme,
    otherTheme: Theme
  }
};
