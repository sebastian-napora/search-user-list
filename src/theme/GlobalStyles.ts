import { createGlobalStyle } from 'styled-components';

import { TSingleTheme } from './types';

export const GlobalStyles = createGlobalStyle<{ theme: TSingleTheme }>`
  body {
    background: ${({ theme }) => theme.colors.body};
    color: ${({ theme }) => theme.colors.text};
    font-family: ${({ theme }) => theme.font.text};
    height: 100vh;
    margin: 0 !important;
  }

  a {
    color: ${({ theme }) => theme.colors.link.text};
    cursor: pointer;
  }

  button {
    border: 0;
    display: inline-block;
    padding: 12px 24px;
    font-size: 14px;
    border-radius: 4px;
    margin-top: 5px;
    cursor: pointer;
    background-color: #1064EA;
    color: #ffffff;
    font-family: ${({ theme }) => theme.font};
  }

  button.btn {
    background-color: ${({ theme }) => theme.colors.button.background};
    color: ${({ theme }) => theme.colors.button.text};
  }
`;
