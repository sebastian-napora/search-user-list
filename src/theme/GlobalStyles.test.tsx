import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import { ThemeProvider } from 'styled-components';

import { GlobalStyles } from './GlobalStyles';

import { THEMES } from './schema';

describe('GlobalStyles', () => {
  it('renders properly every add styles from THEMES', () => {
    const { container, unmount } = render(
      <ThemeProvider {...{ theme: THEMES.data.main }}>
        <GlobalStyles />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();

    unmount();
  });
});
