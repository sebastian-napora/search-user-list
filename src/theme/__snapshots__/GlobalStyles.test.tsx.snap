// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyles renders properly every add styles from THEMES 1`] = `<div />`;
